src/imitation/algorithms/adversarial.py

```diff
@@ -336,29 +321,24 @@ def train(
             self.train_disc(self.disc_batch_size)
             if callback:
                 callback(epoch)
-            logger.dumpkvs()
-
-    def _build_graph(self):
-        # Build necessary parts of the TF graph. Most of the real action happens in
-        # constructors for self.discrim and self.gen_policy.
-        with tf.variable_scope("trainer"):
-            with tf.variable_scope("discriminator"):
-                disc_opt = self._disc_opt_cls(**self._disc_opt_kwargs)
-                self._disc_train_op = disc_opt.minimize(
-                    tf.reduce_mean(self.discrim.disc_loss),
-                    global_step=self._global_step,
-                )
+            logger.dump()
+
+    def _torchify_array(self, ndarray: np.ndarray, **kwargs) -> th.Tensor:
+        return th.as_tensor(ndarray, device=self.discrim.device(), **kwargs)
+
+    def _torchify_with_space(
+        self, ndarray: np.ndarray, space: gym.Space, **kwargs
+    ) -> th.Tensor:
+        tensor = th.as_tensor(ndarray, device=self.discrim.device(), **kwargs)
+        preprocessed = preprocess_obs(
+            tensor,
+            space,
+            # TODO(sam): can I remove "scale" kwarg in DiscrimNet etc.?
```

Comment:
It's intended for image-based observation spaces, which aren't used in any of the tests, etc. I was asking because I don't know whether any downstream code is using it.

Reply:
(I'm inclined to defer this decision until I refactor the whole reward/discriminator abstraction, including state preprocessing)

Reply:
I agree with waiting on this